From 107f7fc484af4dc187d0765653a9403a344fef01 Mon Sep 17 00:00:00 2001
From: PhoebeHui <v-phma@microsoft.com>
Date: Thu, 5 Dec 2019 03:11:36 -0800
Subject: [PATCH 2/2] Configure Jamroot.jam on linux

---
 .../boost-modular-build.cmake                 | 103 +++++++++++++-----
 1 file changed, 74 insertions(+), 29 deletions(-)

diff --git a/ports/boost-modular-build-helper/boost-modular-build.cmake b/ports/boost-modular-build-helper/boost-modular-build.cmake
index 42735be0a..4a3e9211f 100644
--- a/ports/boost-modular-build-helper/boost-modular-build.cmake
+++ b/ports/boost-modular-build-helper/boost-modular-build.cmake
@@ -80,37 +80,82 @@ function(boost_modular_build)
     # endif()
 
     if(VCPKG_CMAKE_SYSTEM_NAME AND NOT VCPKG_CMAKE_SYSTEM_NAME STREQUAL "WindowsStore")
-        set(configure_option)
-        if(DEFINED _bm_BOOST_CMAKE_FRAGMENT)
-            list(APPEND configure_option "-DBOOST_CMAKE_FRAGMENT=${_bm_BOOST_CMAKE_FRAGMENT}")
-        endif()
-        if(DEFINED VCPKG_PYTHON_EXECUTABLE)
-            list(APPEND configure_option "-DVCPKG_PYTHON_EXECUTABLE=${VCPKG_PYTHON_EXECUTABLE}")
-        endif()
-        if(DEFINED VCPKG_PYTHON_INCLUDE)
-            list(APPEND configure_option "-DVCPKG_PYTHON_INCLUDE=${VCPKG_PYTHON_INCLUDE}")
-        endif()
-        if(DEFINED VCPKG_PYTHON_LIBS_RELEASE)
-            list(APPEND configure_option "-DVCPKG_PYTHON_LIBS_RELEASE=${VCPKG_PYTHON_LIBS_RELEASE}")
-        endif()
-        if(DEFINED VCPKG_PYTHON_LIBS_DEBUG)
-            list(APPEND configure_option "-DVCPKG_PYTHON_LIBS_DEBUG=${VCPKG_PYTHON_LIBS_DEBUG}")
+        if(NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "release")
+            set(BOOST_LIB_SUFFIX ${BOOST_LIB_RELEASE_SUFFIX})
+            set(VARIANT "release")
+            set(BUILD_LIB_PATH "lib/")
+            configure_file(${_bm_DIR}/Jamroot.jam ${_bm_SOURCE_PATH}/Jamroot.jam @ONLY)
+            set(configure_option)
+            if(DEFINED _bm_BOOST_CMAKE_FRAGMENT)
+                list(APPEND configure_option "-DBOOST_CMAKE_FRAGMENT=${_bm_BOOST_CMAKE_FRAGMENT}")
+            endif()
+            if(DEFINED VCPKG_PYTHON_EXECUTABLE)
+                list(APPEND configure_option "-DVCPKG_PYTHON_EXECUTABLE=${VCPKG_PYTHON_EXECUTABLE}")
+            endif()
+            if(DEFINED VCPKG_PYTHON_INCLUDE)
+                list(APPEND configure_option "-DVCPKG_PYTHON_INCLUDE=${VCPKG_PYTHON_INCLUDE}")
+            endif()
+            if(DEFINED VCPKG_PYTHON_LIBS_RELEASE)
+                list(APPEND configure_option "-DVCPKG_PYTHON_LIBS_RELEASE=${VCPKG_PYTHON_LIBS_RELEASE}")
+            endif()
+            if(DEFINED VCPKG_PYTHON_LIBS_DEBUG)
+                list(APPEND configure_option "-DVCPKG_PYTHON_LIBS_DEBUG=${VCPKG_PYTHON_LIBS_DEBUG}")
+            endif()
+            if(DEFINED VCPKG_PYTHON_VERSION)
+                list(APPEND configure_option "-DVCPKG_PYTHON_VERSION=${VCPKG_PYTHON_VERSION}")
+            endif()
+            vcpkg_configure_cmake(
+                SOURCE_PATH ${CURRENT_INSTALLED_DIR}/share/boost-build
+                PREFER_NINJA
+                OPTIONS
+                    "-DPORT=${PORT}"
+                    "-DCURRENT_INSTALLED_DIR=${CURRENT_INSTALLED_DIR}"
+                    "-DB2_EXE=${B2_EXE}"
+                    "-DSOURCE_PATH=${_bm_SOURCE_PATH}"
+                    "-DBOOST_BUILD_PATH=${BOOST_BUILD_PATH}"
+                    ${configure_option}
+            )
+            vcpkg_install_cmake()
         endif()
-        if(DEFINED VCPKG_PYTHON_VERSION)
-            list(APPEND configure_option "-DVCPKG_PYTHON_VERSION=${VCPKG_PYTHON_VERSION}")
+
+        if(NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "debug")
+            message(STATUS "Building ${TARGET_TRIPLET}-dbg")
+            set(BOOST_LIB_SUFFIX ${BOOST_LIB_DEBUG_SUFFIX})
+            set(VARIANT debug)
+            set(BUILD_LIB_PATH "debug/lib/")
+            configure_file(${_bm_DIR}/Jamroot.jam ${_bm_SOURCE_PATH}/Jamroot.jam @ONLY)
+            set(configure_option)
+            if(DEFINED _bm_BOOST_CMAKE_FRAGMENT)
+                list(APPEND configure_option "-DBOOST_CMAKE_FRAGMENT=${_bm_BOOST_CMAKE_FRAGMENT}")
+            endif()
+            if(DEFINED VCPKG_PYTHON_EXECUTABLE)
+                list(APPEND configure_option "-DVCPKG_PYTHON_EXECUTABLE=${VCPKG_PYTHON_EXECUTABLE}")
+            endif()
+            if(DEFINED VCPKG_PYTHON_INCLUDE)
+                list(APPEND configure_option "-DVCPKG_PYTHON_INCLUDE=${VCPKG_PYTHON_INCLUDE}")
+            endif()
+            if(DEFINED VCPKG_PYTHON_LIBS_RELEASE)
+                list(APPEND configure_option "-DVCPKG_PYTHON_LIBS_RELEASE=${VCPKG_PYTHON_LIBS_RELEASE}")
+            endif()
+            if(DEFINED VCPKG_PYTHON_LIBS_DEBUG)
+                list(APPEND configure_option "-DVCPKG_PYTHON_LIBS_DEBUG=${VCPKG_PYTHON_LIBS_DEBUG}")
+            endif()
+            if(DEFINED VCPKG_PYTHON_VERSION)
+                list(APPEND configure_option "-DVCPKG_PYTHON_VERSION=${VCPKG_PYTHON_VERSION}")
+            endif()
+            vcpkg_configure_cmake(
+                SOURCE_PATH ${CURRENT_INSTALLED_DIR}/share/boost-build
+                PREFER_NINJA
+                OPTIONS
+                    "-DPORT=${PORT}"
+                    "-DCURRENT_INSTALLED_DIR=${CURRENT_INSTALLED_DIR}"
+                    "-DB2_EXE=${B2_EXE}"
+                    "-DSOURCE_PATH=${_bm_SOURCE_PATH}"
+                    "-DBOOST_BUILD_PATH=${BOOST_BUILD_PATH}"
+                    ${configure_option}
+            )
+            vcpkg_install_cmake()
         endif()
-        vcpkg_configure_cmake(
-            SOURCE_PATH ${CURRENT_INSTALLED_DIR}/share/boost-build
-            PREFER_NINJA
-            OPTIONS
-                "-DPORT=${PORT}"
-                "-DCURRENT_INSTALLED_DIR=${CURRENT_INSTALLED_DIR}"
-                "-DB2_EXE=${B2_EXE}"
-                "-DSOURCE_PATH=${_bm_SOURCE_PATH}"
-                "-DBOOST_BUILD_PATH=${BOOST_BUILD_PATH}"
-                ${configure_option}
-        )
-        vcpkg_install_cmake()
 
         if(NOT EXISTS ${CURRENT_PACKAGES_DIR}/lib)
             message(FATAL_ERROR "No libraries were produced. This indicates a failure while building the boost library.")
-- 
2.25.1

